MC7000 - added notes and replaced pictures (#369)
mixxxdj/manual: source/hardware/controllers/denon_mc7000.rst

The Denon MC7000 is a professional DJ controller which has got 4-channel capability and dual USB connections. These two USB audio interfaces enable two DJs to play together. This controller includes Denon’s high build quality and superior 24-bit audio reproduction, makes this suited to both mobile and club DJs.

.. versionadded:: 2.2.4
.. versionchanged:: 2.3.0

>    Fixed a bug for Vinylmode on Deck 4 and added new features, like library sorting, search through track with Jog Wheel, eject track from Deck, waveform zoom, Fixed Loops, changed Parameter and Censor/Reverse/Spinback button mapping. Added an experimental beat counter (slicer-like).

Reviewer: ```suggestion
   Fixed a vinyl mode bug on deck 4 and added new features, like library sorting, searching through track with jog wheel, ejecting track from deck, waveform zoom, fixed loops and an experimental beat counter (slicer-like) and improved parameter and censor/reverse/spinback button mappings. 
```

Author: Funny you are changing your own wording but I am okay with that too so just copy/paste it.... Cheers!

Reviewer: Hehe, are you sure? "Vinylmode" without a space and with a larger "V" doesn't look like something I'd write. Or maybe I was on my phone and autocorrect kicked in 😁

   **5.4.21**. All kernel versions **5.6.x** and following are supporting the device.
   All recent Linux distributions are shipped with Audio support for the MC7000, like

>   - Ubuntu 20.04 & 20.10

Reviewer 2: I think we can remove the (incomplete) enumeration of all Linux distributions that are known to work. Basically all do, including those that are not listed (NixOS, Fedora, ...). Instead we should only list those that are known **not** to work if there are any. But I don't think this is needed.

Author: You maybe right - that was a requirement a year ago when the ALSA patch just arrived and almost non of the Distributions delivered a Kernel that supported the sound device. I would still keep the information in for the Kernel version starting to support the MC7000

Reviewer 2: Yes, mentioning the kernel version is helpful :+1:

Mapping <https://mixxx.discourse.group/t/denon-mc7000-mapping/18235>`__
thread.

.. _denon_mc7000_uservariables:

User Variables
~~~~~~~~~~~~~~

Please check the :file:`Denon-MC7000-scripts.js` mapping file for user variables to:

> -  activate :ref:`experimental features <denon_mc7000_experimental>` (default: false)

Reviewer: ```suggestion
-  activate :ref:`experimental features <denon_mc7000_experimental>` (default: ``false``)
```

Author: done

   "54", ":hwlabel:`NEEDLE DROP`",                         "Place your finger on a point along this sensor to jump to that point in the track (strip represents entire track)."
   "54", ":hwlabel:`SHIFT` + :hwlabel:`NEEDLE DROP`",      "Press to jump to a position while a track is currently playing."

.. hint::

>   The jog sensitivity and scratch parameters (28), the default Vinyl mode (30), the Pitch Fader ranges (32) as well as the Needle Drop activity (54) can be set by modifying the variables inside the :ref:`Midiscript file<denon_mc7000_uservariables>` accordingly.

Reviewer: ```suggestion
  The jog sensitivity and scratch parameters (28), the default vinyl mode (30), the pitch fader ranges (32) as well as the needle drop activity (54) can be set by modifying the variables inside the :ref:`JavaScript file<denon_mc7000_uservariables>` accordingly.
```

Author: done

   :figclass: pretty-figures

Fixed Loop Mode (yellow LED)
----------------------------

> | This mode lets you set a dedicated loop from current position.
> | Another push on the Pad button will turn the loop off.

Reviewer: ```suggestion
This mode lets you set a dedicated loop from current position.
Another push on the Pad button will turn the loop off.
```

Author: I'd like to keep line breaks in all those instances where I used "|"

   :alt: Denon MC7000 BEATJUMP Mode
   :figclass: pretty-figures

> | Press and hold ":hwlabel:`SHIFT`" and then push a PAD button to jump even more beats.
> | 16, 32, 64 and 128 forward on the upper row and backwards on the lower row.

Reviewer: ```suggestion
Press and hold :hwlabel:`SHIFT` and then push a pad button to jump even more beats.
16, 32, 64 and 128 forward on the upper row and backwards on the lower row.
```

Author: wanna keep line breaks as on the previous comment so that all possible values are not broken between lines. "pad" is now lowercase

.. note::

>    If experimental features were set to `true` in the :ref:`Midiscript file<denon_mc7000_uservariables>` then the PAD LEDs counting the beat AFTER the main CUE point.

Reviewer: ```suggestion
   If experimental features were set to `true` in the :ref:`JavaScript file<denon_mc7000_uservariables>` then the pad LEDs counting the beat *after* the main CUE point.
```

Author: done

Button LEDs are fully mapped for the first function. As you press and hold :hwlabel:`SHIFT` then the secondary functions have only got some flashing LEDs mapped, e.g. TAP and KEY SYNC, when activated.

Press :hwlabel:`SHIFT` + :hwlabel:`Deck` to switch Platter Ring LED Mode.

  - Mode 0 - Single 'off' LED chase (all other LEDs are 'on')
  - Mode 1 - Single 'on' LED chase (all other LEDs are 'off')

.. hint::

>   The default mode can be set to `0` or `1` inside the :ref:`Midiscript file<denon_mc7000_uservariables>`.

Reviewer: ```suggestion
  The default mode can be set to ``0`` or ``1`` inside the :ref:`JavaScript file<denon_mc7000_uservariables>`.
```

Author: done
